connectors/http/http-base/src/main/java/io/camunda/connector/http/base/auth/BasicAuthentication.java

```diff
@@ -22,30 +22,23 @@
 import io.camunda.connector.generator.java.annotation.TemplateProperty;
 import io.camunda.connector.generator.java.annotation.TemplateSubType;
 import jakarta.validation.constraints.NotEmpty;
-import java.util.function.Function;
 
 @TemplateSubType(id = BasicAuthentication.TYPE, label = "Basic")
 public final class BasicAuthentication implements Authentication {
-  @TemplateProperty(ignore = true)
-  private static final String SPEC_PASSWORD_EMPTY_PATTERN = "SPEC_PASSWORD_EMPTY_PATTERN";
-
-  @TemplateProperty(ignore = true)
-  private static final Function<String, String> SPEC_PASSWORD =
-      (psw) -> psw.equals(SPEC_PASSWORD_EMPTY_PATTERN) ? "" : psw;
 
   @FEEL
   @NotEmpty
   @TemplateProperty(group = "authentication")
   private String username;
 
   @FEEL
-  @NotEmpty
   @TemplateProperty(group = "authentication")
   private String password;
```

Comment:
Please add the condition via an element-template generator constraint that this should not be empty. This will render the property as required in the element templates but wont require a `@NotEmpty` constraint on the backend.

Reply:
And please re-generate the element templates for the http base connector based connectors (REST & GraphQL)

Reply:
Thanks, @sbuettner!

Have you seen [Oleksii's comment](https://github.com/camunda/connectors/pull/2061#pullrequestreview-1911664972)?

I'll wait to address your comments here until we can discuss Oleksii's comment.

Reply:
It's probably somewhere, but if not should we add a section (the command basically) in element-template-generator/maven-plugin/README.md describing how to generate the element templates? 
Might be useful if we hire people without prior Maven experience?

Reply:
@ev-codes Yes, I saw it and my proposal would lead to keeping the constraint in the template but removing it from the backend.

Reply:
@johnBgood Good idea, but should be tackled in a separate issue. Please create one, so we dont forget it.

Reply:
Summary:
- @Oleksiivanov agreed with @sbuettner 's approach
- I implemented @sbuettner 's suggested change
- @johnBgood [created the issue here](https://github.com/orgs/camunda/projects/23/views/1?filterQuery=doc&pane=issue&itemId=55227336)